### ISSUE TITLE
Recovery feature broken when double-clicking a project file

### ISSUE THREAD

**Issue by tresf** [LMMS/lmms#4166]
This is likely a MacOS issue and I've tagged it as such.  Please remove the tag if this can be reproduced on Windows or Linux.

Steps to reproduce:
1. Create and save a project
2. Experience a crash
3. Relaunch from the project file (not from the LMMS launcher icon)

Expected behavior:
1. LMMS offers to recover file, click "Recover", recover file opens!

Actual behavior:
1. LMMS offers to recover file, click "Recover", saved project file opens!

I believe this is a race condition with the way [`QApplication` handles the `fileOpen` event specific to MacOS](https://github.com/LMMS/lmms/commit/3dfd9794dbdf9284e2c8a6378139711c1cf7ef06#diff-01990f45b9969995efd402052be03d4bR841).

**Comment by PhysSong** [comment]
How about changing `if ( !queuedFile.isEmpty() ) {` into `if ( !queuedFile.isEmpty() && fileToLoad != recoveryFile ) {`?

**Comment by PhysSong** [comment]
Though I can't test [the possible fix above](https://github.com/LMMS/lmms/issues/4166#issuecomment-365473806), I believe that will fix this issue. @tresf(or someone else) Please test this!

**Comment by tresf** [comment]
@PhysSong I tried a quick test and it failed.

Track I saved:
![screen shot 2018-03-05 at 9 43 12 pm](https://user-images.githubusercontent.com/6345473/37011490-6d59156a-20be-11e8-9a4c-21b1f107632c.png)

Track I lost:
![screen shot 2018-03-05 at 9 42 30 pm](https://user-images.githubusercontent.com/6345473/37011495-73a4dc2e-20be-11e8-8d24-b64b099d963c.png)

Prompt to recover:
![screen shot 2018-03-05 at 9 42 57 pm](https://user-images.githubusercontent.com/6345473/37011522-98b4e8ce-20be-11e8-9109-52c31bc783b3.png)

Track that opened: (track I initially saved, not that which incurred the crash)
![screen shot 2018-03-05 at 9 43 12 pm](https://user-images.githubusercontent.com/6345473/37011490-6d59156a-20be-11e8-9a4c-21b1f107632c.png)

**Comment by PhysSong** [comment]
I thought it's supposed to work, since the change will block the only possibility that can change `fileToLoad`. @tresf Are you sure with the result? If so, then how about `if ( !queuedFile.isEmpty() && !fileToLoad.isEmpty() ) {`?

If the new attempt doesn't work, I guess we need to dig further...

**Comment by tresf** [comment]
> I thought it's supposed to work

Yeah, me too.  I can run some more controlled tests, but this isn't a show-stopper for 1.2.0.  If people's default reaction is to launch LMMS from its launcher, the recovery feature works still, just not from the project file.

**Comment by zonkmachine** [comment]
> Track that opened: (track I initially saved, not that which incurred the crash)

Testing the recovery system easily gets confusing. Are you sure you wait long enough for the auto save to kick in?

**Comment by PhysSong** [comment]
> Testing the recovery system easily gets confusing.

You don't need to wait for auto-save. Manually creating `recover.mmp` in LMMS working directory will also work. A test can be done like this:
1. Copy an project file into LMMS working directory and rename it to `recover.mmp`
2. Open *another* project by double-clicking

**Comment by tresf** [comment]
> Are you sure you waited long enough for the auto save to kick in?

Yes.  It I'd set to one minute. I waited 3.

**Comment by musikBear** [comment]
@PhysSong what content has `recoveryFile` when you call

**Comment by tresf** [comment]
@musikBear thanks for the observation.  Recovery only occurs if `fileToLoad == recoveryFile`, so by process of elimination, @PhysSong's logic should work.  You can see this logic here: https://github.com/LMMS/lmms/blob/3dfd9794dbdf9284e2c8a6378139711c1cf7ef06/src/core/main.cpp#L846

Before we spend too much time, I'll do some more, controlled unit testing as promised.  Unit tests are tricky because they require the `LMMS.app` to be registered with the OS as if it was normally installed.  This is done by a background `lsregister` service and sets up the appropriate `.mmpz` file associations.

Although the old `LMMS.app` was purged, the new `LMMS.app` was compiled and installed, this wouldn't be the first time a build didn't include a fix, so I'll do the proper work on my end first. ;).

**Comment by musikBear** [comment]
>  You can see this logic here: 

👍 ok!
